eng/common/TestResources/New-TestResources.ps1

```diff
@@ -315,6 +320,7 @@ foreach ($templateFile in $templateFiles) {
         "$($serviceDirectoryPrefix)AZURE_AUTHORITY_HOST" = $context.Environment.ActiveDirectoryAuthority;
         "$($serviceDirectoryPrefix)RESOURCE_MANAGER_URL" = $context.Environment.ResourceManagerUrl;
         "$($serviceDirectoryPrefix)SERVICE_MANAGEMENT_URL" = $context.Environment.ServiceManagementUrl;
+        "$($serviceDirectoryPrefix)STORAGE_ENDPOINT_SUFFIX" = $context.Environment.StorageEndpointSuffix;
```

Comment:
I'm intending on adding this to `Azure.Core` as a base `TestEnvironment` member as well;  it will be used across the Messaging SDKs and appears to be hardcoded in many of the other libraries as well.

Reply:
Is there any reason to tie it to Storage by name? Some other services have end points, IIRC, so can this be generic? Or is it really intended to be specific to Storage?

Reply:
This is the storage specific endpoint suffix; as far as I know, it is only applicable to storage.   For context, the Azure Environment looks something like this for the public cloud:

```text
Name                                              : AzureCloud
EnableAdfsAuthentication                          : False
OnPremise                                         : False
ActiveDirectoryServiceEndpointResourceId          : https://management.core.windows.net/
AdTenant                                          : Common
GalleryUrl                                        : https://gallery.azure.com/
ManagementPortalUrl                               : https://go.microsoft.com/fwlink/?LinkId=254433
ServiceManagementUrl                              : https://management.core.windows.net/
PublishSettingsFileUrl                            : https://go.microsoft.com/fwlink/?LinkID=301775
ResourceManagerUrl                                : https://management.azure.com/
SqlDatabaseDnsSuffix                              : .database.windows.net
StorageEndpointSuffix                             : core.windows.net
ActiveDirectoryAuthority                          : https://login.microsoftonline.com/
GraphUrl                                          : https://graph.windows.net/
GraphEndpointResourceId                           : https://graph.windows.net/
TrafficManagerDnsSuffix                           : trafficmanager.net
AzureKeyVaultDnsSuffix                            : vault.azure.net
DataLakeEndpointResourceId                        : https://datalake.azure.net/
AzureDataLakeStoreFileSystemEndpointSuffix        : azuredatalakestore.net
AzureDataLakeAnalyticsCatalogAndJobEndpointSuffix : azuredatalakeanalytics.net
AzureKeyVaultServiceEndpointResourceId            : https://vault.azure.net
AzureOperationalInsightsEndpointResourceId        : https://api.loganalytics.io
AzureOperationalInsightsEndpoint                  : https://api.loganalytics.io/v1
AzureAnalysisServicesEndpointSuffix               : asazure.windows.net
AnalysisServicesEndpointResourceId                : https://region.asazure.windows.net
AzureAttestationServiceEndpointSuffix             : attest.azure.net
AzureAttestationServiceEndpointResourceId         : https://attest.azure.net
```

Reply:
I think as long as we follow a consistent way of turning these names into environment variables we'll be able to avoid duplicating the names across languages. Proposal here: https://github.com/Azure/azure-sdk-tools/issues/757#issuecomment-663349442

I'm not aware of other languages using environment variables for these purposes yet but could see that happening in the future.